File: Java/Easy/Accepted/longest-continuous-increasing-subsequence_7.java
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        
        int n = nums.length;
        if (n < 2) {
            return n;
        }
        int max = 1;
        int l = 0;
        int r = 1;
        while (r < n) {
            if (nums[r] > nums[r - 1]) {
                r++;
            } else {
                max = Math.max(max, r - l);
                l = r;
                r++;
            }
        }
        return Math.max(max, r - l);
    }
}

File: C++/Easy/Accepted/longest-continuous-increasing-subsequence_7.cpp
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        
        int n = nums.size();
        if (n <= 1) return n;
        int res = 1;
        int curr = 1;
        for (int i = 1; i < n; i++) {
            if (nums[i] > nums[i - 1]) {
                curr++;
            } else {
                res = max(res, curr);
                curr = 1;
            }
        }
        res = max(res, curr);
        return res;
    }
};